backend/src/Service/Subscription/SubscriptionService.php

```diff
@@ -0,0 +1,306 @@
+               
+        //         if ($currentSubscription['id'] == $subscription->isCurrent) {
+                    
+        //             $subscription->isCurrent= SubscriptionConstant::SUBSCRIBE_CURRENT;
+        //         }
+               
+        //         else {
+                   
+        //             $subscription->isCurrent = SubscriptionConstant::SUBSCRIBE_NOT_CURRENT;
+        //         }
+               
+        //     }
+
+            $response[] = $this->autoMapping->map(MySubscriptionsResponse::class, MySubscriptionsResponse::class, $subscription);
+           
+        }
+        
+        // dd($response);
+        return $response;
+    }
+  
+    public function subscriptionUpdateState($request)
+    {
+        $result = $this->subscriptionManager->subscriptionUpdateState($request);
+
+        return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $result);
+    }
+
+    public function updateFinish($id, $status)
+    {
+        $result = $this->subscriptionManager->updateFinish($id, $status);
+       
+        return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $result);
+    }
+
+    public function changeIsFutureToFalse($id)
+    {
+        $result = $this->subscriptionManager->changeIsFutureToFalse($id);
+
+        return $this->autoMapping->map(SubscriptionEntity::class, SubscriptionResponse::class, $result);
+    }
+
+    public function getSubscriptionsPending()
+    {
+        $response = [];
+        $items = $this->subscriptionManager->getSubscriptionsPending();
+       
+        foreach ($items as $item) {
+            $response[] = $this->autoMapping->map('array', SubscriptionByIdResponse::class, $item);
+        }
+        return $response;
+    }
+    
+    public function getSubscriptionById($id)
+    {
+        $response = [];
+        $items = $this->subscriptionManager->getSubscriptionById($id);
+      
+        foreach ($items as $item) {
+            $response[] = $this->autoMapping->map('array', SubscriptionByIdResponse::class, $item);
+        }
+        return $response;
+    }
+
+    public function subscriptionIsActive($storeOwner, $subscribeId)
+    {
+        $res = $this->saveFinishAuto($storeOwner, $subscribeId);
+        if($res->carsStatus){
+
+            return SubscriptionConstant::CARS_FINISHED;
+        }
+ 
+        $item = $this->subscriptionManager->subscriptionIsActive($storeOwner, $subscribeId);
+        if ($item) { 
+
+          return  $item['status'];
+        }       
+     }
+
+    // check subscription , if time is finish or order count is finish, change status value to 'finished'
+    public function saveFinishAuto($storeOwner, $subscribeId)
+    {
+        $response = [];
+        //Get full information for the current subscription
+      
+        $remainingOrdersOfPackage = $this->subscriptionManager->getRemainingOrders($storeOwner, $subscribeId);
+        dd($remainingOrdersOfPackage);
```

Comment:
I believe _dd_ statement here should be commented.

Reply:
I haven't used it yet